edb/protocol/messages.py

```diff
@@ -789,6 +797,15 @@ class Execute(ClientMessage):
     expected_cardinality = EnumOf(UInt8, Cardinality,
                                   'Expected result cardinality.')
     command_text = String('Command text.')
+
+    state_typedesc_id = UUID('Updated session state descriptor ID.')
+    state_data = ArrayOf(
```

Comment:
Should this be `FixedArrayOf(1, Dataelement)` then?  Also, what's the rationale behind making this an array?

Reply:
Oh there wasn't a specific reason - in `Data` message it's also an array of `DataElement`.

Reply:
I think there is a bit of a confusion proliferating here.  The `Data` message should not contain an array of `DataElement` either.  The `1` there is _not_ a number of array elements, it's a number of _columns_ in the underlying Postgres tuple.  EdgeDB always generates output tuples with a single element.  I think it will be much less confusing if we simply declare a reserved field in `Data`, and remove `DataElement`.